plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/NugetGalleryFacetImpl.java

```diff
@@ -244,14 +247,24 @@ public void put(final InputStream inputStream) throws IOException, NugetPackageE
       recordMetadata.put(CREATED, creationTime);
       recordMetadata.put(LAST_UPDATED, creationTime);
       recordMetadata.put(PUBLISHED, creationTime);
-
+      OrientVertex component = null;
       try (InputStream in = tempStream.get()) {
-        createOrUpdatePackage(storageTx, recordMetadata, in);
+        component = createOrUpdatePackage(storageTx, recordMetadata, in);
       }
 
-      maintainAggregateInfo(storageTx, recordMetadata.get(ID));
+      String id = recordMetadata.get(ID);
+      String version = recordMetadata.get(VERSION);
+      maintainAggregateInfo(storageTx, id);
 
+      boolean isNew = component.getIdentity().isNew();  // must check before commit
       storageTx.commit();
+
+      if (isNew) {
+        getEventBus().post(new ComponentCreatedEvent(component, getRepository()));
+      }
+      else {
+        getEventBus().post(new ComponentUpdatedEvent(component, getRepository()));
+      }
     }
```

Comment:
-1 for firing events within transaction. As we seen before intersecting orientdb transactions does not work. But if you fire them outside the transaction (@cwilper may confirm) the object vertex will not be usable). So, firing outside transaction may only contain the vertex id and event listeners will have to get the vertex out of db.